adam_core/coordinates/frame.py

```diff
@@ -0,0 +1,17 @@
+import numpy as np
+import pyarrow.compute as pc
+from quivr import StringField, Table
+
+
+# TODO: Replace with DictionaryField or similar
+#       Investigate whether this class is even necessary
```

Comment:
I think we agree that the reason to put them in a class of their own is if we want to attach some custom methods to it, which for the time being we don't have yet. 

What I see us doing, especially as we add more frames like a Galactic frame (needed for visual stuff), is attaching a rotation matrix to each Frame class that rotates it to ecliptic J2000? Kind of like how we define transformations to and from cartesian. Our rotation matrices currently live in constants.py but we could move them to frames.py. I'd need to think a bit about the interface. So maybe this could work for a follow-up PR?